src/Language/PureScript/CodeGen/JS/Common.hs

```diff
@@ -16,6 +16,11 @@ moduleNameToJs (ModuleName pns) =
   let name = intercalate "_" (runProperName `map` pns)
   in if nameIsJsBuiltIn name then "$$" ++ name else name
 
+properToJs :: String -> String
+properToJs name
+  | nameIsJsReserved name || nameIsJsBuiltIn name = "$$" ++ name
+  | otherwise = concatMap identCharToString name
```

Comment:
Might be worth refactoring `identToJs` to use this in the `(Ident name)` case - as the implementation is the same.